src/explore-education-statistics-common/src/services/applicationInsightsService.ts

```diff
@@ -38,3 +41,91 @@ export function initApplicationInsights(
 export function getApplicationInsights(): ApplicationInsights {
   return appInsights;
 }
+
+const REDACTED_VALUE = '__redacted__';
+const REDACTED_TOKENS = [
+  'code',
+  'email',
+  'phone',
+  'token',
+  'passphrase',
+  'password',
+];
+
+/* eslint-disable no-param-reassign */
+export function filterSensitiveData(telemetry: ITelemetryItem): boolean {
+  switch (telemetry.baseType) {
+    case 'RemoteDependencyData':
+      if (telemetry.baseData?.name) {
+        telemetry.baseData.name = filterUri(telemetry.baseData.name);
+      }
+
+      if (telemetry.baseData?.target) {
+        telemetry.baseData.target = filterUri(telemetry.baseData.target);
+      }
+
+      break;
+
+    case 'MetricData':
+      if (telemetry.data?.PageUrl) {
+        telemetry.data.PageUrl = filterUri(telemetry.data.PageUrl);
+      }
+
+      break;
+
+    case 'PageviewData':
+    case 'PageviewPerformanceData':
+      if (telemetry.baseData?.uri) {
+        telemetry.baseData.uri = filterUri(telemetry.baseData.uri);
+      }
+
+      if (telemetry.baseData?.refUri) {
+        telemetry.baseData.refUri = filterUri(telemetry.baseData.refUri);
+      }
+
+      break;
+
+    default:
+      return true;
+  }
+
+  return true;
+}
+/* eslint-enable no-param-reassign */
+
+function filterUri(uri: string): string {
+  const [path, query] = uri.split('?', 2);
```

Comment:
No JS out-of-the-box one-liners to get the query params out of a URI?  Oh, or is the point that you still want to try to redact invalid URLs?

Reply:
Same as above